### ISSUE TITLE
SSE doesn't produce multiple events

### ISSUE THREAD

**Issue by eli-jordan** [ReactiveX/RxNetty#484]
Based on this [example](https://github.com/ReactiveX/RxNetty/blob/0.5.x/rxnetty-examples/src/main/java/io/reactivex/netty/examples/http/sse/HelloSseServer.java) I was unable to get RxNetty to produce multiple events.

The output is

```
data: Interval => 0
data: Interval => 1
data: Interval => 2
data: Interval => 3
data: Interval => 4
data: Interval => 5
data: Interval => 6
data: Interval => 7
data: Interval => 8
data: Interval => 9
...
```

According to the spec there must a blank line in order for the client to dispatch the message, so since there are no blank lines all data is considered part of the same message, and is not dispatched. Is this the intended behaviour? If so, how do I generate a blank line separator to have the message dispatched?

**Comment by NiteshKant** [comment]
@eli-jordan You are right, the data field should have a double new line character ([currently it has just one](https://github.com/ReactiveX/RxNetty/blob/0.5.x/rxnetty-http/src/main/java/io/reactivex/netty/protocol/http/sse/server/ServerSentEventEncoder.java#L103)). However, it should only be for the data attribute and not for other attributes, as I understand from the spec.
Do you want to send a quick PR for this?

**Comment by ArekCzarnik** [comment]
@NiteshKant I have workaround, but this should not be the default case: 

``` java
       return response.transformToServerSentEvents()
                .writeAndFlushOnEach(kafkaEventServer.consume()
                        .throttleLast(500, TimeUnit.MILLISECONDS)
                        .onBackpressureDrop().map(s -> withEventId(Unpooled.buffer().writeBytes(Long.toString(s.offset()).getBytes())
                                , Unpooled.buffer().writeBytes(Bytes.concat(s.message(), "\n".getBytes())))));
    }
```

**Comment by eli-jordan** [comment]
@NiteshKant Sure, I'll take a look at implementing PR.

Since a single event can have multiple `data: ...` lines, do you think it would make sense to have `ServerSentEvent` allow a `List<String>` to be passed?

**Comment by NiteshKant** [comment]
@ArekCzarnik yes if you add a new line at the end of the message, it will have the desired result, it is an easy workaround.

@eli-jordan Is there any semantic difference between breaking the data into multiple data attributes? Does SSE have any max attribute value length limitation?

**Comment by eli-jordan** [comment]
The only real difference based on my reading of the spec, is that if you want to have new lines in message data you need to use multiple `data: ...` lines. However, it looks like the code is splitting on new lines, so `Arrays.asList("foo", "bar")` is equivalent to `"foo\nbar"`. So I don't think the List is required

**Comment by ArekCzarnik** [comment]
Hi @eli-jordan,

are you have change something? If not i would make pull request with simple fix.

Thanks,
Arek

**Comment by eli-jordan** [comment]
I found this bug when evaluating libraries... We ended up choosing something else, and I haven't had time to come back to this. Please feel free to make the change.
